ion/unmarshal_test.go

```diff
@@ -537,3 +537,28 @@ func TestDecode(t *testing.T) {
 	test("()", []interface{}{})
 	test("(1 + two)", []interface{}{1, "+", "two"})
 }
+
+func TestDecodeLotsOfInts(t *testing.T) {
```

Comment:
Could you help me understand how this test exercises the case where `Read()` would return fewer bytes than were requested?

Reply:
Under the covers we create a bufio.Reader with a default (4096-byte) internal buffer. After reading 4 bytes of BVM and 511 x 8 byte integer values, there are 4 bytes remaining in that buffer, plus another 4 in the underlying stream. Pre-fix, we’d do a partial read of those four bytes as a (corrupt) 512th integer, then try to read the remaining four bytes as another value.

(Might be worth trying to add a more specific test with a custom Reader so it’s more obvious what’s going on, if the bufio.Reader doesn’t get in the way? This is the code I wrote to debug the issue and figured I might as well include :))

Reply:
Ah, perfect! Thank you. Would you mind adding a comment to the test explaining the buffer size math?

> Might be worth trying to add a more specific test with a custom Reader so it’s more obvious what’s going on, if the bufio.Reader doesn’t get in the way?

Agreed. I'll open an issue to track this.